frontend/src/homePage/CodeGroups.jsx

```diff
@@ -0,0 +1,124 @@
+import { useState, useEffect } from "react";
+import "./CodeGroups.css";
+import axios from "axios";
+
+function CodeGroups() {
+  const [allGroups, setAllGroups] = useState([]);
+  const [groupName, setGroupName] = useState("");
+  const [members, setMembers] = useState([]);
+  const [memberInput, setMemberInput] = useState("");
+  const [error, setError] = useState("");
+  const [count, setCount] = useState(0);
+
+  const BACKEND_URL = import.meta.env.VITE_BACKEND_URL;
+
+  useEffect(() => {
+    setAllGroups([]);
+    const token = localStorage.getItem("jwt");
+    if (!token) {
+      return;
+    }
+
+    const getGroups = async () => {
```

Comment:
Similar feedback as [week 2](https://github.com/izuchukwumba/flixster/issues/16#:~:text=I%20would%20recommend%20not%20defining%20functions%20inside%20another%20function%2C%20like%20you%20do%20here.), I would recommend not defining functions inside another function. This can lead to long functions and more convoluted code, and here it prevents you from being able to reuse the function to fetch the groups data outside of the useEffect.

Reply:
Fixed. Thanks